lib/BackbeatProducer.js

```diff
@@ -9,110 +9,75 @@ const { withTopicPrefix } = require('./util/topic');
 const KafkaBacklogMetrics = require('./KafkaBacklogMetrics');
 const Constants = require('./constants');
 
-// waits for an ack for messages
-const REQUIRE_ACKS = 'all';
-// time in ms. to wait for acks from Kafka
-const ACK_TIMEOUT = 5000;
-// max time in ms. to wait for flush to complete
-const FLUSH_TIMEOUT = 5000;
-
-/**
-* Backbeat replication topic is currently setup with a config
-* max.message.bytes of 5MB. Producers need to update their
-* messageMaxBytes to get at least 5MB put in the Kafka topic, adding a
-* little extra bytes of padding for approximation.
-*/
+const DEFAULT_POLL_INTERVAL = 2000;
 const PRODUCER_MESSAGE_MAX_BYTES = 5000020;
+const FLUSH_TIMEOUT = 5000;
 
-const CLIENT_ID = 'BackbeatProducer';
 
 class BackbeatProducer extends EventEmitter {
 
-    /**
-    * constructor
-    * @param {Object} config - config
-    * @param {string} [config.topic] - Kafka topic to write to when
-    * using BackbeatProducer.send() calls
-    * @param {boolean} [config.keyedPartitioner=true] - if no
-    * partition is set, tell whether the producer should use keyed
-    * partitioning or the default partitioning of the kafka client
-    * @param {Object} config.kafka - kafka connection config
-    * @param {string} config.kafka.hosts - kafka hosts list
-    * as "host:port[,host:port...]"
-    */
     constructor(config) {
         super();
 
-        const configJoi = joi.object({
-            kafka: joi.object({
-                hosts: joi.string().required(),
-            }).required(),
-            topic: joi.string(),
-            keyedPartitioner: joi.boolean().default(true),
-            pollIntervalMs: joi.number().default(2000),
-            messageMaxBytes: joi.number().default(PRODUCER_MESSAGE_MAX_BYTES),
-        });
-        const validConfig = joi.attempt(config, configJoi,
-                                        'invalid config params');
-        const { kafka, topic, pollIntervalMs, messageMaxBytes } = validConfig;
+        const validConfig = joi.attempt(
+            config, this.getConfigJoi(),
```

Comment:
```suggestion
            config, 
            this.getConfigJoi(),
```
Kinda hard to see there are 3 params here instead of two when two are on one line

Reply:
Was this missed?

Reply:
Sorry, I haven't been able to get back to this until today :sweat: